salt/client/ssh/shell.py

```diff
@@ -26,6 +26,11 @@
 KEY_VALID_RE = re.compile(r'.*\(yes\/no\).*')
 SSH_PRIVATE_KEY_PASSWORD_PROMPT_RE = re.compile(r'Enter passphrase for key', re.M)
 
+# sudo prompt is used to recognize sudo prompting for a password and should therefore be
+# fairly recognizable and unique
+SUDO_PROMPT = '[salt:sudo:d11bd4221135c33324a6bdc09674146fbfdf519989847491e34a689369bbce23]passwd:'
+SUDO_PROMPT_RE = re.compile(r'{0}'.format(SUDO_PROMPT), re.M)
```

Comment:
Can you explain a bit what is going on here?

Reply:
SUDO_PROMPT is the value sudo will output if the user should enter its password (when it is used with the -p parameter). You'll want it to be quite unlike an application will output, so you can recognize it in the output. Really almost any string would do, so I just made something up.

Reply:
Creative. :wink: Why not reuse existing `RSTR` instead?

```python
SUDO_PROMPT = 'salt:sudo:{}...'.format(RSTR)
```

Reply:
I don't think you can do that. SUDO_PROMPT_RE is used in shell._run_cmd() to recognize the sudo password request. In the same method RSTR_RE is used to recognize that we are receiving results. If RSTR was part of SUDO_PROMPT, RSTR_RE might trigger first in stead of SUDO_PROMPT_RE. It is a lot safer for SUDO_PROMPT to be a completely separate unique string.

Reply:
If used the same regexps yes. OK, let's leave it for now as is, but I think this should be revisited a while later.